### PR TITLE
Feature/query api

### PR DESCRIPTION
**Pull request by ezer-dnv**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -53,6 +53,7 @@ services:
             - "8812:8812"
             - "9000:9000"
             - "9003:9003"
+            - "9009:9009"
```
Comment (kristoffersaastad) [inline docker-compose.yml]: This is for influxDb, not used atm

File: docker-compose-development.yml [hunk 2]
```diff
@@ -9,6 +9,7 @@ services:
             - "8812:8812"
             - "9000:9000"
             - "9003:9003"
+            - "9009:9009"
```
Comment (kristoffersaastad) [inline docker-compose-development.yml]: This is for influxDb, not used atm

File: src/QueryApi/Controllers/QueryController.cs [hunk 3]
```diff
@@ -1,22 +1,28 @@
...
     /// <summary>
-    /// Creates a vessel for the current user.
+    /// Search for data channels based in the given filters
     /// </summary>
-    /// <param name="vessel"></param>
-    /// <param name="cancellationToken"></param>
-    [HttpGet(Name = "Query")]
-    public async ValueTask<ActionResult> Query(Guid vessel, CancellationToken cancellationToken)
+    /// <param name="filter"></param>
+    [HttpPost]
```
Comment (kristoffersaastad) [inline src/QueryApi/Controllers/QueryController.cs]: Add enpoint name? Mabye 'query'?

File: src/QueryApi/Models/DataChannelDto.cs [hunk 4]
```diff
@@ -1,42 +1,43 @@
...
-    double? FormatRestriction_MinInclusive,
-    int? FormatRestriction_MinLength,
-    int? FormatRestriction_Pattern,
-    int? FormatRestriction_TotalDigits,
-    string? FormatRestriction_WhiteSpace,
-    DateTime Timestamp
-);
+public class DataChannelDto
+{
+    public string? DataChannelId { get; set; }
```
Comment (kristoffersaastad) [inline src/QueryApi/Models/DataChannelDto.cs]: Why is this optional? This, and other required values

File: src/QueryApi/Models/DataChannelResponse.cs [hunk 5]
```diff
@@ -0,0 +1,32 @@
+namespace VistaLab.QueryApi.Models
+{
+    public sealed record DataChannelResponse
+    {
+        public string? Query { get; set; }
+        public Column[]? Columns { get; set; }
+        public List<object[]>? Dataset { get; set; }
```
Comment (kristoffersaastad) [inline src/QueryApi/Models/DataChannelResponse.cs:7]: What is Dataset compared to DataChannels?

**Review by kristoffersaastad — COMMENTED** [review]
Really nice! I only have a few comments/questions

File: src/QueryApi/Models/DataChannelDto.cs (hunk 4 above)
Comment (ezer-dnv) [reply src/QueryApi/Models/DataChannelDto.cs]: To not get this error:

CS8618	Non-nullable property 'DataChannelId' must contain a non-null value when exiting constructor. Consider declaring the property as nullable.

https://docs.microsoft.com/en-us/dotnet/csharp/nullable-references#nullable-contexts

Unless we change our project config

**Review by kristoffersaastad — APPROVED** [review]
Good job

**Pull request merged**